src/bindings/c/ov/src/ov_c_api.cpp

```diff
@@ -0,0 +1,1310 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <iterator>
+#include <string>
+#include <utility>
+#include <cassert>
+#include <map>
+#include <vector>
+#include <set>
+#include <algorithm>
+#include <chrono>
+#include <tuple>
+#include <memory>
+#include <streambuf>
+#include <fstream>
+
+#include "c_api/ov_c_api.h"
+#include "openvino/openvino.hpp"
+
+struct ov_core {
+    std::shared_ptr<ov::Core> object;
+};
+
+struct ov_node {
+    std::shared_ptr<ov::Node> object;
+};
+
+struct ov_output_node {
+    std::shared_ptr<ov::Output<const ov::Node>> object;
```

Comment:
I think we don't need to hold `Output` objects wrapped in a shared pointer. Is there a reason to use it this way in this struct?

Reply:
Not other reason, just it will be convenient if use shared_ptr.

Reply:
done